mmpose/core/post_processing/compact_heatmap.py

```diff
@@ -0,0 +1,82 @@
+import numpy as np
+
+
+def compact_heatmap(heatmap, threshold=3e-3):
+    """Convert Heatmap into a more compact format. The input heatmap is a numpy
+    arrayof the shape H x W. The function will find a tight rectangle that
+    contain all positions with score > threshold, then crop the heatmap with
+    the tight rectangle. The output heatmap will also be converted to float16.
+    To restore the original heatmap, a quadruple (x, y, original_w, original_h)
+    of type int16 will also be returned.
+
+    Args:
+        heatmap (np.ndarray([H, W])): The input heatmap.
+        threshold (float): The heatmap score threshold.
+
+    Returns:
+        tuple: New Heatmap and Crop Info.
+        - new_heatmap (np.ndarray([h, w])): The cropped heatmap.
+        - quadruple (np.ndarray([4])): The crop info.
+    """
+
+    h, w = heatmap.shape
+    y, x = np.where(heatmap > threshold)
+
+    # In which case not a single element is larger than threshold
+    if x.shape[0] == 0:
+        new_heatmap = np.zeros([0, 0], dtype=np.float16)
+        return new_heatmap, np.array([0, 0, w, h], dtype=np.int16)
+    min_x, max_x = min(x), max(x)
+    min_y, max_y = min(y), max(y)
+    new_heatmap = heatmap[min_y:max_y, min_x:max_x]
```

Comment:
Double check if this is needed: `max_y` -> `max_y+1`

Reply:
It's OK to use `max_y` here since the shape can always match (we just record the left-up corner of the compact heatmap and restore the heatmap with the coordinates).